tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/StackTraceLeak/HttpResponseIntegration.cs

```diff
@@ -0,0 +1,55 @@
+// <copyright file="HttpResponseIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+#if NETFRAMEWORK
+
+using System;
+using System.ComponentModel;
+using System.Web;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.Configuration;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.StackTraceLeak;
+
+/// <summary>
+/// HttpResponseIntegration integration
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "System.Web",
+    TypeName = "System.Web.HttpResponse",
+    ParameterTypeNames = new[] { ClrNames.Exception, ClrNames.Bool },
+    MethodName = "WriteErrorMessage",
+    ReturnTypeName = ClrNames.Void,
+    MinimumVersion = "4.0.0",
+    MaximumVersion = "4.*.*",
+    IntegrationName = nameof(IntegrationId.StackTraceLeak),
+    InstrumentationCategory = InstrumentationCategory.Iast)]
+
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public static class HttpResponseIntegration
+{
+    /// <summary>
+    /// OnMethodBegin callback
+    /// </summary>
+    /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+    /// <param name="exception">The exception to be shown.</param>
+    /// <param name="dontShowSensitiveErrors">The dontShowSensitiveErrors parameter of WriteErrorMessage.</param>
+    /// <typeparam name="TTarget">Type of the target</typeparam>
+    /// <returns>Calltarget state value</returns>
+    internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, Exception exception, bool dontShowSensitiveErrors)
+    {
+        if (HttpRuntime.UsingIntegratedPipeline && !dontShowSensitiveErrors)
+        {
+            return StackTraceLeakIntegrationCommon.OnExceptionLeak(IntegrationId.StackTraceLeak, exception);
+        }
+
+        return CallTargetState.GetDefault();
+    }
```

Comment:
I'm wondering whether `UsingIntegrationPipeline` is enough here 🤔 In `HttpContextHelper.AddHeaderTagsFromHttpResponse()` (and other places) we have to add an additional `try-catch`, because we still run into issues accessing the `HttpContext`. My gut feeling is that we _don't_ need to do the following, because I don't think you're trying to access the response headers anyway (inside `IastModule` for example). 

Just dropping it here for context, just in case you think it _could_ be an issue 🙂 



```suggestion
    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(HttpResponseIntegration));
    private static bool _canReadHttpResponseHeaders = true;

    internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, Exception exception, bool dontShowSensitiveErrors)
    {
        try
        {
            if (_canReadHttpResponseHeaders && HttpRuntime.UsingIntegratedPipeline && !dontShowSensitiveErrors)
            {
                return StackTraceLeakIntegrationCommon.OnExceptionLeak(IntegrationId.StackTraceLeak, exception);
            }
        }
        catch (PlatformNotSupportedException ex)
        {
            // Despite the HttpRuntime.UsingIntegratedPipeline check, we can still fail to access response headers, for example when using Sitefinity: "This operation requires IIS integrated pipeline mode"
            Log.Error(ex, "Unable to access response headers performing StackLeak detection. Disabling for the rest of the application lifetime.");
            _canReadHttpResponseHeaders = false;
        }
        return CallTargetState.GetDefault();
    }
```